added view all tags to the website

NathanCenter opened the pull request:
#### Changes Made
1. Added TagControler.cs, ` to `Controllers ` directory.
2. Added Tag.cs, ` to `Models` directory.
3.  Added ITagRepository.cs, TagRepository.cs ` to `Repository` directory.
4. added TagMangement.js and TagView.js to Modules directory

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout nc-displayAllTags
    npm start
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. Test app functionality.
   When you first log into the website and run your API you should see Tag Manager in the nav bar. When you click on that it should send you to a new page that shows a list of all the tags that are listed in the datebase.
expected outcome is.
    When user clicks on tag manager, then a new page should show with a list of all the tags.
6. View the code file.
   closes #35

matt-rich-anderson reviewed (approved):
I review the functionality and everything seems to work as expected.

heymonicakay commented on Tabloid/client/src/modules/TagView.js:
```diff
@@ -0,0 +1,32 @@
+import { getAllTags } from './TagManagment';
```
This is a component module, so it should not be in the modules directory. Please move this to the components directory.

NathanCenter replied:
Do I need to move TagManagement to modules or TagView ?

heymonicakay commented on Tabloid/client/src/components/Header.js:
```diff
@@ -28,6 +29,9 @@ export default function Header({ isLoggedIn }) {
               </NavItem>
             }
+               <NavItem>
+                <NavLink tag={RRNavLink} to="/Tags" >Tag Managment</NavLink>
```
Make sure to keep your casing consistent... you've got pascal casing here, but in your app views module you've got lowercase

heymonicakay reviewed (changes requested):
Please see my comments. Also, please reference the issue tickets related to this PR in your PR description.

NathanCenter commented:
Fixed changes that Monica made  showed me

1. This is a component module, so it should not be in the modules directory. Please move this to the components directory.
Added that TagManagement.js is now in the modules directory.
3. Make sure to keep your casing consistent... you've got pascal casing here, but in your app views module you've got lowercase
 changed the path from lower case to uppercase on ApplicationViews.js

heymonicakay commented on Tabloid/Controllers/TagController.cs:
```diff
@@ -0,0 +1,24 @@
+using Microsoft.AspNetCore.Mvc;
+using Tabloid.Repositories;
+
+namespace Tabloid.Controllers
```
Should this controller be auth protected? do we need any other auth checks on the route?

matt-rich-anderson reviewed (approved):
Ito looks like you have made Monica's requested changes

heymonicakay commented on Tabloid/Repositories/TagRepository.cs:
```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using Tabloid.Models;
+
+namespace Tabloid.Repositories
```
Are there DbUtils methods that could be used in this repo to do some of this work? We'll want to use those utils methods consistently across the codebase.

heymonicakay commented on Tabloid/client/src/components/TagManagment.js:
```diff
@@ -0,0 +1,16 @@
+const baseUrl = 'api/Tag';
```
This file does not a component. This file should be in the modules directory.

NathanCenter replied:
@heymonicakay Hey I changed the tag management and move it to modules

heymonicakay commented on Tabloid/client/src/modules/TagView.js:
```diff
@@ -0,0 +1,32 @@
+import { getAllTags } from '../components/TagManagment';
```
This file contains a component. This file should be in the components directory.

NathanCenter replied:
@heymonicakay  Hey I changed the tagView and moved it to the components directory

heymonicakay commented on Tabloid/client/src/components/TagManagment.js:
```diff
@@ -0,0 +1,16 @@
+    .then((res) => res.json())
+};
+
+export const addTag = (tag) => {
```
is addTag supported by your API yet? if not, remove this function from this PR.

NathanCenter replied:
@heymonicakay got rid of add tag on the tag mangment.js

heymonicakay commented on Tabloid/client/src/components/TagManagment.js:
```diff
@@ -0,0 +1,16 @@
+const baseUrl = 'api/Tag';
+
+export const getAllTags = () => {
```
if this route is auth protected, you'll need to get an api token to pass in to this fetch call.

NathanCenter replied:
@heymonicakay  Added api token class to the tag mangement!

heymonicakay reviewed (changes requested):
Please see my comments
